src/parser/jobs/drg/modules/DrgBuffWindow.tsx

```diff
@@ -0,0 +1,210 @@
+import React, {Fragment} from 'react'
+import {MessageDescriptor} from '@lingui/core'
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {CastEvent, BuffEvent} from 'fflogs'
+import {Icon, Message} from 'semantic-ui-react'
+
+import ACTIONS, {Action} from 'data/ACTIONS'
+import {Status} from 'data/STATUSES'
+import {Data} from 'parser/core/modules/Data'
+import Module, {dependency} from 'parser/core/Module'
+import Suggestions, {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {Timeline} from 'parser/core/modules/Timeline'
+import {ActionLink} from 'components/ui/DbLink'
+import {RotationTable, RotationTableTargetData, RotationTableNotesMap} from 'components/ui/RotationTable'
+
+// generally, you're not supposed to use drg buffs immediately after these
+const BAD_BUFF_ACTIONS = [
+	ACTIONS.CHAOS_THRUST.id,
+	ACTIONS.FULL_THRUST.id,
+]
+
+class BuffWindowHistory {
+	start: number
+	end?: number
+	casts: CastEvent[] = []
+	isBad: boolean
+
+	private data: Data
+
+	constructor(data: Data, start: number, isBad: boolean) {
+		this.start = start
+		this.isBad = isBad
+		this.data = data
+	}
+
+	get gcds(): number {
+		return this.casts
+			.map(e => this.data.getAction(e.ability.guid))
+			.filter(a => a && a.onGcd)
+			.length
+	}
+}
+
+// the main reason that this doesn't extend core buff windows is that DRG
+// needs a few extra hooks to determine if the buff had an optimal placement,
+// which requires us to know which GCDs were used immediately before the buff
```

Comment:
Given the reasoning of "don't use right after one of the 3d hit skills" and "include 8 GCDs per window", does that mean the expectation is that each buff window should include a total of 2 3d hit skills (1 of each in single target, I would assume, but if multi-dotting is a thing then maybe make the good skills check a little more forgiving to take 2 Chaos Thrust instead of one of each)?  Would that be a better presentation to end users rather than "don't use it right after the big hits" and/or allow for simplifying this logic and directly extending core BuffWindow?

Reply:
I believe the current thinking wants each window to include CT, FT, and FC/WT, so I think Lance Charge could be swapped pretty easily to core. Dragon Sight needs to be able to track two statuses and add an additional hook for checking solo DS use (but that check could easily go into a different module), so the core BuffWindow would need a slight adjustment for that to work.

We should get a comment from @Rietty or other DRG theorycrafters about which suggestion format would be better (I'm leaning towards switching to core myself).

Reply:
Per our discussion on Discord, I’m going to go ahead and merge this and we can move forward with investigating alternatives for the buff windows separately.